File: RaspberryPi/example/c/DigitalReadWrite.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <sys/stat.h>
#include <sys/types.h>
#include <mqueue.h>
#include <unistd.h>

#include "../../src/UUGear.h"

int main(int argc, char **argv)
{
	setupUUGear();
	
	setShowLogs(1);
	
	UUGearDevice dev = attachUUGearDevice ("UUGear-Arduino-7853-2668");
	
	if (dev.fd != -1)
	{
		setPinModeAsOutput (&dev, 13);
		int i = 0;
		for (i = 0; i < 5; i ++)
		{
			setPinHigh (&dev, 13);
			
			usleep(200000);
			
			setPinLow (&dev, 13);
		
			usleep(200000);
		}
		
		setPinModeAsInput (&dev, 9);
		
		printf("Pin 9 status=%d\n", getPinStatus(&dev, 9));
		
		sleep(1);
		
		detachUUGearDevice (&dev);
	}
	else
	{
		printf("Can not open UUGear device.\n");	
	}

	cleanupUUGear();
		
    return 0;
}

File: RaspberryPi/example/python/DigitalReadWrite.py
from time import sleep
from UUGear import *

UUGearDevice.setShowLogs(0)

device = UUGearDevice('UUGear-Arduino-7853-2668')

if device.isValid():
	device.setPinModeAsOutput(13)
	for i in range(5):
		device.setPinHigh(13)
		sleep(0.2)
		device.setPinLow(13)
		sleep(0.2)
	
	device.setPinModeAsInput(9)
	print 'Pin 9 status=', device.getPinStatus(9)
	
	device.detach()
	device.stopDaemon()
else:
	print 'UUGear device is not correctly initialized.'